src/systemd/sd-dhcp-lease.h

```diff
@@ -24,6 +24,8 @@
 #include <netinet/in.h>
 #include <sys/types.h>
 
+/* #include "dns-resolver.h" FIXME: whynot? */
+typedef struct ResolverData ResolverData;
```

Comment:
sd-dhcp is supposed to be a library we export for other apps. We follow a different naming scheme for all code like that, i.e. call objects sd_xyz in all lowercase.

also, all structures must be opaque, and fields only accessible via accessors (i.e. no open coded structs). And we don't cheat on that, not even if consumer is in our codebase.

or in other words: this should be called `sd_dhcp_resolver_data` or so and have accessors for the various fields. Exactly how this is done for sd_dhcp_lease

Reply:
Gotcha, thanks for the explanation.